File: app/src/main/java/com/nawin/androidmvparchitecture/MvpModule.java
package com.nawin.androidmvparchitecture;

import android.content.Context;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by brainovation on 7/20/17.
 */
@Module
public class MvpModule {
    private final Context context;

    MvpModule(MVPApp app) {
        this.context = app;
    }

    @Provides
    Context provideContext() {
        return context;
    }

    @Provides
    @Singleton
    Gson gson() {
        return new GsonBuilder().create();
    }
}

File: app/src/main/java/com/nawin/androidmvparchitecture/MvpModule.kt
package com.nawin.androidmvparchitecture

import android.content.Context
import com.google.gson.Gson
import com.google.gson.GsonBuilder
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Created on 11/29/17.
 */
@Module
class MvpModule(var app: MVPApp) {
    @Provides
    internal fun provideContext(): Context = app

    @Provides
    @Singleton
    internal fun gson(): Gson = GsonBuilder().create()
}